src/Analyzer.JsonRuleEngine/Operators/InequalityOperator.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Templates.Analyzer.RuleEngines.JsonEngine.Operators
+{
+    /// <summary>
+    /// An operator that evaluates the "greater", "greaterOrEquals", "less", and "lessOrEquals" JSON expressions.
+    /// </summary>
+    internal class InequalityOperator : LeafExpressionOperator
+    {
+        /// <inheritdoc/>
+        public override string Name => GetName();
+
+        /// <summary>
+        /// Whether the operator also considers equality
+        /// </summary>
+        public Boolean OrEquals;
+
+        /// <summary>
+        /// Creates an InequalityOperator.
+        /// </summary>
+        /// <param name="specifiedValue">The value specified in the JSON rule.</param>
+        /// <param name="isNegative">Whether the operator compares by greater than or by less than.</param>
```

Comment:
Can you rename the `isNegative` parameter to something more descriptive (like `greaterThan`, etc.)?  It can still use the IsNegative class member internally to store it.